CRM/Queue/Queue.php

```diff
@@ -60,7 +62,17 @@ public function __construct($queueSpec) {
    */
   public function isActive(): bool {
     $status = CRM_Core_DAO::getFieldValue('CRM_Queue_DAO_Queue', $this->_name, 'status', 'name', TRUE);
-    // Note: In the future, we may want to incorporate other data (like maintenance-mode or upgrade-status) in deciding active queues.
+    if ($status === 'active' && \Civi::settings()->get('queue_suspended')) {
```

Comment:
@eileenmcnaughton I'm not certain if this is an issue for you or not, but I did some `r-run`, and it didn't quite work the way I expected

The issue is that `Civi::settings()` is designed to cache the setting. But we have multiple concurrent workers (pipe-sessions); each worker has a separate cache. Eventually [workers expire](https://lab.civicrm.org/dev/coworker/-/blob/v0.2/src/Configuration.php?ref_type=tags#L22-55) -- so eventually, the cache is reset. But I would expect the flag to be more immediate.

Pushing up a patch...

Reply:
thanks @totten